src/pages/OfficeView.tsx

```diff
@@ -1,46 +1,37 @@
-import { useCallback, useEffect, useState } from 'react'
+import { useState } from 'react'
 import { Card, Col, FormControl, InputGroup, Row } from 'react-bootstrap'
 import { Link, useLoaderData, useNavigate } from 'react-router-dom'
 import { useTranslation } from 'react-i18next'
 import OfficeInfo from 'src/components/OfficeInfo'
 import RatingsBadge from 'src/components/RatingsBadge'
 import { OfficeLoaderReturn } from 'src/loaders/officeLoader'
-import Agent from 'src/types/Agent'
 import businessIcon from 'src/images/business.svg'
 import { ApiAgent } from 'src/api'
-import Paginator from 'src/components/Paginator'
-import { debounce } from 'lodash-es'
-import SearchMeta from 'src/types/SearchMeta'
+import { useSuspenseInfiniteQuery } from '@tanstack/react-query'
+import AnimatedList from 'src/components/AnimatedList'
+import { InView } from 'react-intersection-observer'
 
 export default function OfficeView() {
   const { office } = useLoaderData() as OfficeLoaderReturn
   const [searchValue, setSearchValue] = useState('')
   const navigate = useNavigate()
   const { t } = useTranslation()
-  const [agents, setAgents] = useState<Agent[]>([])
-  const [meta, setMeta] = useState<SearchMeta>()
 
-  const getAgents = useCallback(
-    async (page: number) => {
-      return await ApiAgent.list({
-        officeId: office.id,
-        search: searchValue,
-        page,
-      })
-    },
-    [office.id, searchValue],
-  )
-
-  useEffect(() => {
-    const initializeAgents = async (page: number) => {
-      const agentData = await getAgents(page)
-      setAgents(agentData.data)
-      setMeta(agentData.meta)
-    }
-    const handleSearchInput = debounce(initializeAgents, 500)
+  const { data, fetchNextPage, hasNextPage, isFetching } =
+    useSuspenseInfiniteQuery({
+      queryKey: ['searchVal', searchValue],
+      queryFn: async ({ pageParam = 0 }) =>
+        await ApiAgent.list({
+          officeId: office.id,
+          search: searchValue,
+          page: pageParam as number,
+        }),
+      getNextPageParam: ({ meta }) =>
+        meta.page < meta.totalPages - 1 ? meta.page + 1 : undefined,
+      initialPageParam: 0,
```

Comment:
Since the list of agents for a particular office isn't likely to change often, I think we should actually add a `staleTime` here so results get cached. Lets start with 5 minutes: `staleTime: 1000 * 60 * 5` .

Reply:
Top of this page has a quick read on cachine and staleTime for context: https://tanstack.com/query/v4/docs/framework/react/guides/important-defaults.

After adding staleTime, If you filter your network tab to "agents" requests, and enter a search term in the UI, delete it, and enter it again, see that it doesn't need to request the data a second time. The second time you enter the search term, it looks it up and finds it in the cache. Hence the `searchValue` portion of the cache key.
<img width="1276" alt="image" src="https://github.com/ProjectProtocol/project-protocol-web/assets/6488787/4699da2c-fe7c-4d67-92d0-347244851f28">